components/clp-package-utils/clp_package_utils/scripts/compress.py

```diff
@@ -21,28 +25,165 @@
+
+    # fmt: off
+    compress_cmd = [
+        "python3",
+        "-m", "clp_package_utils.scripts.native.compress",
+        input_type,
+        "--config", str(config_path)
+    ]
+    # fmt: on
+    if parsed_args.timestamp_key is not None:
+        compress_cmd.append("--timestamp-key")
+        compress_cmd.append(parsed_args.timestamp_key)
+    if parsed_args.tags is not None:
+        compress_cmd.append("--tags")
+        compress_cmd.append(parsed_args.tags)
+    if parsed_args.no_progress_reporting is True:
+        compress_cmd.append("--no-progress-reporting")
+
+    if InputType.FS == input_type:
+        pass
+    elif InputType.S3 == input_type:
+        aws_access_key_id = parsed_args.aws_access_key_id
+        aws_secret_access_key = parsed_args.aws_secret_access_key
+        if parsed_args.aws_credentials_file:
+            aws_access_key_id, aws_secret_access_key = parse_aws_credentials_file(
+                pathlib.Path(parsed_args.aws_credentials_file)
+            )
+        if aws_access_key_id and aws_secret_access_key:
+            compress_cmd.append("--aws-access-key-id")
+            compress_cmd.append(aws_access_key_id)
+            compress_cmd.append("--aws-secret-access-key")
+            compress_cmd.append(aws_secret_access_key)
+    else:
+        raise ValueError(f"Unsupported input type: {input_type}.")
+
+    compress_cmd.append("--target-list")
+    compress_cmd.append(str(target_list_path))
+
+    return compress_cmd
+
+
+def _add_common_arguments(
+    args_parser: argparse.ArgumentParser, default_config_file_path: pathlib.Path
+) -> None:
     args_parser.add_argument(
         "--config",
         "-c",
         default=str(default_config_file_path),
         help="CLP package configuration file.",
     )
-    args_parser.add_argument("paths", metavar="PATH", nargs="*", help="Paths to compress.")
-    args_parser.add_argument(
-        "-f", "--path-list", dest="path_list", help="A file listing all paths to compress."
-    )
     args_parser.add_argument(
         "--timestamp-key",
         help="The path (e.g. x.y) for the field containing the log event's timestamp.",
     )
     args_parser.add_argument(
         "-t", "--tags", help="A comma-separated list of tags to apply to the compressed archives."
     )
+    args_parser.add_argument(
+        "--no-progress-reporting", action="store_true", help="Disables progress reporting."
+    )
+
+
+def _validate_fs_input_args(
+    parsed_args: argparse.Namespace,
+    args_parser: argparse.ArgumentParser,
+) -> None:
+    # Validate some input paths were specified
+    if len(parsed_args.paths) == 0 and parsed_args.path_list is None:
+        args_parser.error("No paths specified.")
+
+    # Validate paths were specified using only one method
+    if len(parsed_args.paths) > 0 and parsed_args.path_list is not None:
+        args_parser.error("Paths cannot be specified on the command line AND through a file.")
+
+
+def _validate_s3_input_args(
```

Comment:
I think the two `ValueError`s in this file could be `argparse.error` invocations?